frame/support/procedural/src/storage_alias.rs

```diff
@@ -0,0 +1,524 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation of the `storage_alias` attribute macro.
+
+use frame_support_procedural_tools::generate_crate_access_2018;
+use proc_macro2::{Span, TokenStream};
+use quote::{quote, ToTokens};
+use syn::{
+	ext::IdentExt,
+	parenthesized,
+	parse::{Parse, ParseStream},
+	punctuated::Punctuated,
+	token, Attribute, Error, Ident, Result, Token, Type, TypeParam, Visibility,
+};
+
+/// Represents a path that only consists of [`Ident`] separated by `::`.
+struct SimplePath {
```

Comment:
I find myself often wanting to look for these convenience syntax structs as well, IIRC I had written something similar in `construct_runtime` as well. Maybe we should try collecting them into a module/crate so that we can all make good use of it? Something like an extension of `syn` but for FRAME? `syn-frame`?

Reply:
Yeah sounds like a good idea. Maybe we just use the `procedural-tools` that already contains some "tooling" (I'm open to rename this crate)

Reply:
That works. I'm not sure if I can think of a better name, since the other one that I can come up with right now is `utils`, but we have an overabundance of crates that are called `utils` already...